### PR TITLE
Track missing dartdocs

### PR DESCRIPTION
**Pull request by Hixie**
This will help catch when we accidentally break --dartdocs.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @yjbanov, @tvolkert

File: dev/devicelab/bin/tasks/dartdocs.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      otherLines += 1;
+    }
+    final int result = await analysis.exitCode;
+    clock.stop();
+    if (publicMembers == 0 && otherErrors == 0 && result != 0)
+      throw new Exception('flutter analyze exited with unexpected error code $result');
+    if (publicMembers != 0 && otherErrors != 0 && result == 0)
+      throw new Exception('flutter analyze exited with successful status code despite reporting errors');
+    if (otherLines != 2)
+      throw new Exception('flutter analyze had unexpected output ($otherLines unexpected line${ otherLines == 1 ? "" : "s" })');
```
Comment (tvolkert) [inline dev/devicelab/bin/tasks/dartdocs.dart]: If we expect 2, and the number is 3, then it's technically not "3 unexpected lines"

File: dev/devicelab/bin/tasks/dartdocs.dart [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      print('analyzer stdout: $entry');
+      otherLines += 1;
+    }
+    final int result = await analysis.exitCode;
+    clock.stop();
+    if (publicMembers == 0 && otherErrors == 0 && result != 0)
+      throw new Exception('flutter analyze exited with unexpected error code $result');
+    if (publicMembers != 0 && otherErrors != 0 && result == 0)
+      throw new Exception('flutter analyze exited with successful status code despite reporting errors');
+    if (otherLines != 2)
```
Comment (tvolkert) [inline dev/devicelab/bin/tasks/dartdocs.dart]: Why 2?  Should this be a constant, or maybe just add cases for the two lines in the stdio processing above?

File: dev/devicelab/bin/tasks/dartdocs.dart [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      }
+    }
+    await for (String entry in analysis.stdout.transform(UTF8.decoder).transform(const LineSplitter())) {
+      print('analyzer stdout: $entry');
+      otherLines += 1;
+    }
+    final int result = await analysis.exitCode;
+    clock.stop();
+    if (publicMembers == 0 && otherErrors == 0 && result != 0)
+      throw new Exception('flutter analyze exited with unexpected error code $result');
```
Comment (tvolkert) [inline dev/devicelab/bin/tasks/dartdocs.dart]: Why not just if `result != 0`?  Or at least return `TaskResult.failure()` down below if `result != 0`?

Comment (Hixie) [inline dev/devicelab/bin/tasks/dartdocs.dart]: It should always return != 0, unless we're in the rare state of having documented every public member.

Comment (tvolkert) [inline dev/devicelab/bin/tasks/dartdocs.dart]: ack

**Review by tvolkert — APPROVED** [review]
LGTM modulo the comment about the magic number 2

File: dev/devicelab/bin/tasks/dartdocs.dart (hunk 2 above)
Comment (Hixie) [inline dev/devicelab/bin/tasks/dartdocs.dart]: done

File: dev/devicelab/bin/tasks/dartdocs.dart (hunk 1 above)
Comment (Hixie) [inline dev/devicelab/bin/tasks/dartdocs.dart]: fixed as a result of the other

**Comment by Hixie** [comment]
will check in on green unless i hear otherwise

**Pull request merged**